### ISSUE TITLE
XBox Game Carbon Emission calculator

### ISSUE THREAD

**Issue by Sachidanand-s** [Green-Software-Foundation/hack#80]
### Prize category

Best Plugin

### Overview

To capture the carbon emission of Xbox games we will develop the tool, based on the Green Software Foundation methodology called SCI (Software Carbon Intensity) and IF (Impact Framework), will capture carbon emissions across the Xbox game lifecycle: measuring power consumption during gameplay, estimating server and download emissions, raising awareness of development impact, utilizing "carbon-aware" features, and promoting responsible recycling. 

### Have you got a project team yet?

Yes and we are still open to extras members

### Terms of Participation

- [X] I agree to the hackathon [Rules & Terms](https://github.com/Green-Software-Foundation/hack/wiki/Rules-and-Terms) and [Code of Conduct](https://github.com/Green-Software-Foundation/hack/wiki/Code-of-Conduct)

**Comment by srini1978** [comment]
Will this tool be deployed on Xbox console to capture  energy and other metadata during  the activities you mentioned - gameplay etc

**Comment by Sachidanand-s** [comment]
> Will this tool be deployed on Xbox console to capture energy and other metadata during the activities you mentioned - gameplay etc

Yes that is also option we are looking and also looking other options as well as we have XBox lab for all such testing so how we can deploy it on the console ..

**Comment by jawache** [comment]
Thanks @Sachidanand-s!

There are two approaches here I think.

1- Measuring the emissions end to end of a title with say an SCI score of carbon / user.

- manifest file which captures all the components involved in running that title for a week, all cloud, network and Xbox usage. Perhaps a month's worth of data with daily buckets.
- capture users playing the title every day.
- in the manifest file compute the carbon for each component and use the sci plugins to compute carbon /user for the whole month.
- we have some plugins that help but the process of writing that manifest file will uncover the gaps and inform you which plugins you'll need to write to compute the manifest. For example a perhaps a plugin which computes energy consumed by an Xbox for a given time/duration/utilisation.

2- measure just the title locally on the Xbox with a benchmark process.
- run a benchmark on the Xbox (robot playing the game for 10hrs)
- capture observations like ultilzation.
- generate a manifest file containing all the data.
- build any extra Plugins you need, e.g. the same one I mentioned above to get energy/embodied for an Xbox.
- the manifest can be executed to calculate an sci score for the title 

Looking forward to seeing what comes out of this! The games sector is something I get asked about a lot but have very few answers for people 😄

**Comment by Sachidanand-s** [comment]
Thanks for the details @jawache  ... 
You have articulated it very well and in details .. And exactly that's how we are thinking to capture carbon emission of the Games from the consoles and Cloud services as well. 

We really want to deploy some background job on the Console to capture the carbon emission of games based on SCI method. And the second part we want to capture the carbon emission is being emitted by the cloud services at the same time while game is playing.  

Regarding the PlugIns we need to explore it further that how we can build the custom manifest file along with the custom PlugIns for game and need some help in this area ..

**Comment by Sophietn** [comment]
Hi @Sachidanand-s 
I see you have the 'recruiting' label for your Hackathon project!
Would you be interested in taking a segment on any of our Weekly Live Hackathon Prep and Q&A Sessions taking place every Monday 2.30 GMT, to discuss your idea and attract participants?

**Comment by Sachidanand-s** [comment]
Hi @Sophietn , Sure we will do it .. I am in PST Time Zone so might not be able to join at 2.30 pm GMT but will align one of Architect from our team side to join it and share our point of view ..

**Comment by Sophietn** [comment]
Hi @Sachidanand-s, glad you've accepted! 

First step is to please [register your project](https://hack.greensoftware.foundation/register/) and ask any existing team members to register at the same link, under the same team name. 
From here we'll get your email address to communicate directly there. 

Would someone be available to join next Monday 11th March 2.30pm GMT? 

cc The hosts of our Monday livestreams: @jawache and @jmcook1186

**Comment by Sophietn** [comment]
Hi @Sachidanand-s 
Are you still looking for additional team participants?
Would you like to attend our Launch event on the 18th March 2.30pm GMT, to discuss your project idea and attract team members?

**Comment by sachidanand** [comment]
Hi @Sophietn , yes we are looking for team members.. I will add my team members to attend the Launch call and discuss the ideas there..

**Comment by Sophietn** [comment]
Great thank you. Please [register your project](https://hack.greensoftware.foundation/register/). This way I'll have your email address to invite you to the Live stream event launch Monday 18th 2.30PM GMT. 
Please also ask your team mates to [register](https://hack.greensoftware.foundation/register/) - thanks!

**Comment by Sophietn** [comment]
Hi @sachidanand if you'd like to receive an invitation to the the webinar today, please [register your project](https://hack.greensoftware.foundation/register/). Thank you

**Comment by SKushwaha1** [comment]
Hi, updating this thread on behalf of @sachidanand

Overview
To capture the carbon emission of Xbox games we will develop the tool, based on the Green Software Foundation methodology called SCI (Software Carbon Intensity) and IF (Impact Framework), will capture carbon emissions across the Xbox game lifecycle: measuring power consumption during gameplay, estimating server and download emissions, raising awareness of development impact, utilizing "carbon-aware" features, and promoting responsible recycling.

Here is the architectural diagram for the proposed tool, focusing primarily on three nodes for calculating the SCI of Xbox:

1. Xbox Cloud
2. Network
3. Client Device

![image](https://github.com/Green-Software-Foundation/hack/assets/26564852/35bbf636-ce09-4459-ba58-805adf03cefe)

I am slso attaching the Input/ Output Yaml file for better understanding 

![image](https://github.com/Green-Software-Foundation/hack/assets/26564852/3f10b77d-cd39-4095-a408-07f4e6dc8594)

**Comment by sachidanand** [comment]
> Hi @Sachidanand-s please don't forget to register your project: https://hack.greensoftware.foundation/register/
> 
> This provides you direct access to the Impact Framework team for your questions and also benefits from our community partners (Microsoft & Electricity Maps).
> 
> **You must register your project before you can submit your solution for judging.**

Hi @russelltrow , I have already registered the project so could you check if it there in your repo ?
